docs/protocol/avs.mdx

```diff
@@ -0,0 +1,246 @@
+StrategyManager->>Strategy1: token.transferFrom(sender, strategy, amount)
+StrategyManager->>DelegationManager: increaseDelegatedShares
+```
+
+In Eigenlayer's design, staker funds are safeguarded as they are not made accessible to the AVS until slashing is activated. This ensures that no funds are at risk prematurely. Additionally, this design does not allow users to selectively opt into the Hyperlane AVS alone, as staking is managed through the StrategyManager and directed towards a specific operator.
+
+### Workflow for reading stake from an ISM
+
+```mermaid
+sequenceDiagram
+  box Destination Chain
+    participant EconomicISM
+  end
+
+  actor StakeOracle
+
+  box Ethereum L1
+    participant ECDSAStakeRegistry
+    participant DelegationManager
+  end
+
+  StakeOracle->>ECDSAStakeRegistry: quorum().strategies
+  StakeOracle->>DelegationManager: getOperatorShares(operator, strategies)
+  StakeOracle->>EconomicISM: updateStake()
+```
+
+Stake weight given to different operators will be determined by the aggregate stake delegated to them on the L1. There's several ways of obtaining this but the most straightforward way will be to use the native L1->L2 bridge if the destination chain is a rollup.
+
+:::note
+
+The EconomicISM is not yet live in production and the above sequence diagram can change based on the final implementation.
+
+:::
+
+### Workflow for slashing
+
+```mermaid
+sequenceDiagram
+  box Origin Chain
+    actor Watcher
+    participant NativeChallenger
+  end
+
+  box Ethereum L1
+    participant IRemoteChallenger
+    participant HSM
+    participant Slasher
+  end
+
+
+Watcher->>NativeChallenger: challenge(address, bytes32,bytes32)
+NativeChallenger->>IRemoteChallenger: "postChallenge(operator)" via nativeBridge
+IRemoteChallenger->>HSM: handleChallenge(operator)
+HSM-->>HSM: checkIfEnrolled(operator)
+HSM->>Slasher: freezeOperator(operator)
+```
+
+The slashing mechanism is designed to be permissionless and flexible. The `IRemoteChallenger` interface allows for different implementations of the slashing mechanism. As an example, the `postChallenge` function is called by the native challenger on the origin chain, which then calls the `handleChallenge` function on the `HyperlaneServiceManager` contract. The `HyperlaneServiceManager` contract then checks if the operator is enrolled in the challenger and calls the `freezeOperator` function on the `Slasher` contract to freeze the operator's stake.
+
+:::note
+
+Both the challenger and slasher contracts are not yet live in production. This sequence diagram can change based on the final implementation.
+
+:::
+
+### Protocol SLA
+
+Anyone can join Hyperlane as a validator, i.e., there's no defacto validator set. You can start attesting to messages from your chain but you'll only be rewarded if you are picked by an [ISM](./ISM/modular-security.mdx) deployer for verification. Similarily, you can only get slashed by the said ISM deployer if you enroll into their Challenger.
```

Comment:
Sometimes I would read this as a 1:1 correspondence between ISM deployer and challenger. I dont expect that to be true, so I would be consistent and just say that you will be slashed by the challenger.

Reply:
Ditto re: rewards, would just omit ISM selection. At the top I would just be explicit that enrollment and operation does not mean that the signature actually secure messages until setup in an ISM